libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs

```diff
@@ -504,6 +504,11 @@ private async Task<DialogTurnResult> CallGenerateAnswerAsync(WaterfallStepContex
             var result = new List<QueryResult>();
             if (response.Answers.Any())
             {
+                if (response.Answers.First() != null && response.Answers.First().AnswerSpan != null && !string.IsNullOrEmpty(response.Answers.First().AnswerSpan.Text)) 
```

Comment:
each time calling First() will iterate through list. 

also, we no need to do this logic here

Reply:
changes excluded.